openghg/store/_metstore.py

```diff
@@ -119,14 +140,30 @@ def _store(self, met_data: METData) -> None:
         from openghg.store.base import Datasource
 
         metadata = met_data.metadata
+        
+        from openghg.store import assign_data, datasource_lookup, load_metastore
+        metastore = load_metastore(key=self._metakey)
+        
+        met = METStore.load()
+
 
         datasource = Datasource()
         datasource.add_data(metadata=metadata, data=met_data.data, data_type="met")
         datasource.save()
+        
 
         date_str = f"{metadata['start_date']}_{metadata['end_date']}"
 
         name = "_".join((metadata["site"], metadata["network"], date_str))
         self._datasource_uuids[datasource.uuid()] = name
+        
+        print(self._datasource_uuids[datasource.uuid()], datasource.uuid())
+
+        met.add_single_datasource(uuid={name:datasource.uuid()}, data=met_data, metastore=metastore)
+
+        met.save()
+        #metastore.close()
```

Comment:
When using the metadata store it should either be used with a context manager or explicitly closed, records won't get written correctly otherwise.